packs/email/sensors/imap_sensor.py

```diff
@@ -101,7 +135,57 @@ def _process_message(self, uid, mailbox):
             'subject': subject,
             'message_id': message_id,
             'body': body,
-            'attachments': None
+            'has_attachments': has_attachments,
+            'attachments': []
         }
 
+        if has_attachments and download_attachments:
+            self._logger.debug('[IMAPSensor]: Downloading attachments for message {}'.format(uid))
+            result = self._download_and_store_message_attachments(message=message)
+            payload['attachments'] = result
+
         self._sensor_service.dispatch(trigger=self._trigger, payload=payload)
+
+    def _download_and_store_message_attachments(self, message):
+        """
+        Method which downloads the provided message attachments and stores them in a datasatore.
+
+        :rtype: ``list`` of ``dict``
+        """
+        attachments = message.attachments
```

Comment:
Yeah, sadly that's how the underlying library works - the attachment is still downloaded, but we don't store it in the datastore.

Not ideal, but we can look into optimizing it (not downloading the attachment) later.